django/river/services.py

```diff
@@ -70,14 +70,11 @@ def preview(mapping: dict, resource_id: str, primary_key_values: Optional[list])
             documents.append(document)
             resource_type = document.get("resourceType")
             try:
-                construct_fhir_element(resource_type, document)
-            except ValidationError as e:
-                errors.extend(
-                    [
-                        f"{err['msg'] or 'Validation error'}: "
-                        f"{e.model.get_resource_type()}.{'.'.join([str(l) for l in err['loc']])}"
-                        for err in e.errors()
-                    ]
-                )
+                validation_response = fhir_api.validate(resource_type, document, fhir_api_auth_token)
+                errors.extend(validation_response.get("issue"))
+            except Exception as e:
+                # format error in the same way as OperationOutcome.issue
+                # https://www.hl7.org/fhir/operationoutcome-definitions.html#OperationOutcome.issue
+                errors.append({"severity": "error", "code": "", "diagnostics": str(e)})
```

Comment:
So does `validate` raise or not?

Reply:
it raises if the response is not JSON-serializable (ie: it's not an OperationOutcome error)

Reply:
I find this a bit weird. What about catching the error in `fhir_api.validate` and not having this try/except?

Reply:
better now ?